MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -4240,7 +4241,7 @@ public static string GetHwTonemapFilter(EncodingOptions options, string hwTonema
 
             if (doDeintH2645)
             {
-                var deintFilter = GetHwDeinterlaceFilter(state, options, "videotoolbox");
+                var deintFilter = supportsHwDeint ? GetHwDeinterlaceFilter(state, options, "videotoolbox") : GetSwDeinterlaceFilter(state, options);
```

Comment:
SW deint filter cannot be used after the hwupload filter.

Also you have to handle the case that HW decoder is enabled but HW encoder is disabled, in which case it should fall back to the legacy pipeline.